sdk/offchain/src/identifier.rs

```diff
@@ -0,0 +1,386 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::subaddress::{Subaddress, SubaddressParseError};
+use anyhow::{anyhow, Result};
+use bech32::{self, u5, FromBase32, ToBase32};
+use diem_sdk::{
+    move_types::account_address::AccountAddressParseError,
+    types::{
+        account_address::AccountAddress,
+        account_config::allowed_currency_code_string,
+        chain_id::{ChainId, NamedChain},
+    },
+};
+use std::collections::HashMap;
+use thiserror::Error;
+use url::Url;
+
+const DIEM_BECH32_VERSION: u8 = 1;
+
+/// Intent is a struct holdind data decoded from Diem Intent Identifier string
+/// https://dip.diem.com/dip-5/#format
+#[derive(Ord, PartialOrd, Eq, PartialEq, Hash, Clone)]
+pub struct Intent {
+    pub account_address: AccountAddress,
+    pub subaddress: Subaddress,
+    pub currency_code: Option<String>,
```

Comment:
Lets try and use the Currency type from the sdk

Reply:
https://github.com/diem/diem/blob/9c941ad07eb7148532ee71c88c987c341e434b6c/sdk/src/transaction_builder.rs#L523-L526

The `Currency` type only has two entries that are hard coded, but we'll have additional currencies dynamically added via Identifiers:

https://github.com/diem/diem/blob/9c941ad07eb7148532ee71c88c987c341e434b6c/types/src/account_config/constants/diem.rs#L62-L73

Will discuss offline.

Reply:
AFAIK there are no plans to add additional currencies in the short term and only AOS has the ability to add currencies to the system so there are only those 2 "hardcoded" currencies which are supported.